src/libraries/System.Text.Json/tests/Serialization/ExtensionDataTests.cs

```diff
@@ -250,6 +250,115 @@ public static void IgnoredDataShouldNotBeExtensionData()
+        {
+            public override Dictionary<string, object> Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void Write(Utf8JsonWriter writer, Dictionary<string, object> value, JsonSerializerOptions options)
+            {
+                writer.WriteString("MyCustomOverflowWrite", "OverflowValueWrite");
+            }
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_SupportsWritingToCustomSerializerWithOptions()
+        {
+            var root = new ClassWithExtensionData
+            {
+                Overflow = new Dictionary<string, object>
+                {
+                    ["TestKey"] = "TestValue",
+                },
+            };
+
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new DictionaryOverflowConverter());
+
+            string json = JsonSerializer.Serialize(root, options);
+            Assert.Equal(@"{""MyCustomOverflowWrite"":""OverflowValueWrite""}", json);
+        }
+
+        private class ClassWithExtensionDataWithAttributedConverter
+        {
+            [JsonExtensionData]
+            [JsonConverter(typeof(DictionaryOverflowConverter))]
+            public Dictionary<string, object> Overflow { get; set; }
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_SupportsWritingToAttributedCustomSerializer()
+        {
+            var root = new ClassWithExtensionDataWithAttributedConverter
+            {
+                Overflow = new Dictionary<string, object>
+                {
+                    ["TestKey"] = "TestValue",
+                },
+            };
+
+            string json = JsonSerializer.Serialize(root);
+            Assert.Equal(@"{""MyCustomOverflowWrite"":""OverflowValueWrite""}", json);
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_IgnoresCustomSerializerWithOptions()
+        {
+            var options = new JsonSerializerOptions();
+            options.Converters.Add(new DictionaryOverflowConverter());
+
+            ClassWithExtensionData obj
+                = JsonSerializer.Deserialize<ClassWithExtensionData>(@"{""TestKey"":""TestValue""}", options);
+
+            Assert.Equal("TestValue", ((JsonElement)obj.Overflow["TestKey"]).GetString());
+        }
+
+        [Fact]
+        public static void ExtensionPropertyObjectValue_IgnoresExplicitCustomSerializer()
+        {
+            ClassWithExtensionDataWithAttributedConverter obj
+                = JsonSerializer.Deserialize<ClassWithExtensionDataWithAttributedConverter>(@"{""TestKey"":""TestValue""}");
+
+            Assert.Equal("TestValue", ((JsonElement)obj.Overflow["TestKey"]).GetString());
+        }
+
+        private class ClassWithJsonElementExtensionData
```

Comment:
We need another class where the extension property doesn't have the converter attribute and a test (for serialization and deserialization) where the converter is added at runtime with `options.Converter.Add`.

Reply:
For completeness, we should also add tests (serialization and deserialization) for a class where the extension property dictionary is a custom type, and the `JsonConverterAttribute` is applied on the custom type itself.

Reply:
The new test covers them